### ISSUE TITLE
AZ Phewas portal linkouts and changes to widget

### ISSUE THREAD

**Issue by buniello** [opentargets/issues#3321]
The [AstraZeneca PheWAS Portal](https://azphewas.com/) team have shared their gene and disease links that we can use for our pages link-outs. This happened though personal comms with @d0choa.

- **Format for the gene links:**

https://azphewas.com/geneView/{dataset}/{gene}/glr/{type}
 
Where:
- dataset is either:
the 450k (v4) release: edd5b6b3-622d-418c-9313-572d154054cf
or the latest 470k (v5) release: ba08a93f-501e-44e6-a332-98ce2f852279
- gene: The original gene id from the data, eg. BRCA1
- type: The “binary” or “continuous” suffix depending on whether the table in OpenTargets is showing odds ratio (binary) or beta (continuous). Without the suffix it will always default to the binary traits.

**BRCA1 example in 470k**: https://azphewas.com/geneView/ba08a93f-501e-44e6-a332-98ce2f852279/BRCA1/glr/binary

- **Disease links:**

The team shared the full list in a spreadsheet

- The team will also share **variant links** and package with qualifying variants (cases and controls) for significant tests.

**Comment by buniello** [comment]
@ireneisdoomed and myself discussed this in person and shared relevant material. Irene let me know if you have any questions for me. 
From my side, i will think about what's he best way to visualise their gene links.

**Comment by buniello** [comment]
Discussed and decided:

-  not to include AZ phewas portal gene links to the gene burden widget, only phenotype links within the `Cohort/Project` column
- remove `study ID` column
- remove GWAS Catalog link from widget header

**Comment by ireneisdoomed** [comment]
The data contains the information to build both links based on the provided tables.

1. **Link to their phenotype page**
This is directly available in the first element of `urls.url`.
For example: https://azphewas.com/phenotypeView/ba08a93f-501e-44e6-a332-98ce2f852279/dd036e4f-7ca4-47ec-9bd3-7fd1d3f45ac9/
My recommendation is that we update the link in the `Cohort/Project` column of each AZ evidence to this one.

2. **Link to their gene page**
This can be built concatenating several fields. The pattern is: `https://azphewas.com/geneView/{releaseVersion}/{targetFromSourceId}/`
For example: https://azphewas.com/geneView/ba08a93f-501e-44e6-a332-98ce2f852279/A1BG/
My recommendation is that we update the `AstraZeneca PheWAS Portal` link in the subheader of the widget since it will remain constant for all AZ rows in that widget.

Just double checking, neither `urls` nor `releaseVersion` were used for the burden evidence so far, but they were there for other sources. We don't need to change the API, right?

**Comment by d0choa** [comment]
@ireneisdoomed No need to change the API if the fields were already in the schema for another evidence data source. The API treats all evidence as a single dataset with a single schema independently of the data source.

**Comment by buniello** [comment]
Tagging @gjmcn for above comments

**Comment by gjmcn** [comment]
@ireneisdoomed Should the `releaseVersion` be available with the current version of the platform? If I go to the BRCA1 associations page, expand the first Gene Burden evidence circle (breast neoplasm) and tweak the API query to include `releaseVersion` in the rows, I get `null`.

**Comment by ireneisdoomed** [comment]
That's right. This is a new field so it's empty in the data in production

**Comment by d0choa** [comment]
@gjmcn `releaseVersion` is expected to be populated only when `projectId` == `AstraZeneca PheWAS Portal`. Only in these cases we are expecting to build the links

Query based on today's `dev` version: 

```
query query {
  disease(efoId: "EFO_0003869") {
    id
    geneBurdenSummary: evidences(
      ensemblIds: ["ENSG00000012048"]
      enableIndirect: true
      datasourceIds: ["gene_burden"]
      size: 10
    ) {
      count
      rows {
        studyId
        projectId
        releaseVersion
      }
    }
  }
}

```

```
{
  "data": {
    "disease": {
      "id": "EFO_0003869",
      "geneBurdenSummary": {
        "count": 71,
        "rows": [
          {
            "studyId": null,
            "projectId": "REGENERON",
            "releaseVersion": null
          },
          {
            "studyId": null,
            "projectId": "REGENERON",
            "releaseVersion": null
          },
          {
            "studyId": null,
            "projectId": "REGENERON",
            "releaseVersion": null
          },
          {
            "studyId": null,
            "projectId": "AstraZeneca PheWAS Portal",
            "releaseVersion": "ba08a93f-501e-44e6-a332-98ce2f852279"
          },
          {
            "studyId": null,
            "projectId": "REGENERON",
            "releaseVersion": null
          },
          {
            "studyId": null,
            "projectId": "AstraZeneca PheWAS Portal",
            "releaseVersion": "ba08a93f-501e-44e6-a332-98ce2f852279"
          },
          {
            "studyId": null,
            "projectId": "REGENERON",
            "releaseVersion": null
          },
          {
            "studyId": null,
            "projectId": "AstraZeneca PheWAS Portal",
            "releaseVersion": "ba08a93f-501e-44e6-a332-98ce2f852279"
          },
          {
            "studyId": null,
            "projectId": "REGENERON",
            "releaseVersion": null
          },
          {
            "studyId": null,
            "projectId": "AstraZeneca PheWAS Portal",
            "releaseVersion": "ba08a93f-501e-44e6-a332-98ce2f852279"
          }
        ]
      }
    }
  }
}
```

**Comment by gjmcn** [comment]
Sorry, confused myself! - was working in dev but used the production GraphQL playground.

**Comment by gjmcn** [comment]
@buniello @d0choa

For the AZ PheWAS link in the widget subheader, if there are no rows where `projectId == AstraZeneca PheWAS Portal` should we still include the link but just point to `https://azphewas.com`?

Seems more consistent to always include the same links in the subheader and AZ PheWAS can (or will always?) have a page for the target. To get the direct link in this case will be more work though - which I can look into if worth it.

**Comment by buniello** [comment]
Hi @gjmcn let's disscuss this in person tomorrow. Thank you for this 👍🏼

**Comment by buniello** [comment]
@gjmcn please let me know when this is ready to test

**Comment by gjmcn** [comment]
@carcruz @chinmehta I put in a [PR for this issue](https://github.com/opentargets/ot-ui-apps/pull/391) but @buniello cannot test it because the Netlify preview was cancelled - clicking on details shows that initializing failed:

> Failed during stage 'checking build content for changes': Canceled build due to no content change  

Have you seen this before? Is there an easy fix?

**Comment by carcruz** [comment]
Hi @gjmcn, @buniello 

Here, you can check the preview: https://deploy-preview-391--ot-platform.netlify.app

@gjmcn I just "re-try" to run the build in the UI. This happens because the changes in the PR occurred in files out of the `app/platform` folder. In order to fix this, we will need a smarter dev environment

**Comment by buniello** [comment]
both links, disease (in widget) and gene (source link in widget header) look good 👍

**Comment by d0choa** [comment]
It would be good to fix a minor issue that got lost in limbo. In the rare cases in which `projectId` can be `null` it would be good not to show the parenthesis. The issue was already in production. 

From PKD2 - chronic kidney disease associations:

![Screenshot 2024-06-17 at 10 41 09](https://github.com/opentargets/issues/assets/5097586/cab1e0e6-54dd-464a-81d7-68f634c9da7a)

**Comment by gjmcn** [comment]
Latest commit fixes this.

**Comment by prashantuniyal02** [comment]
Released in Platform 24.06
